clickhouse-native-jdbc/src/main/java/com/github/housepower/jdbc/ClickHouseConnection.java

```diff
@@ -305,7 +317,43 @@ public static ClickHouseConnection createClickHouseConnection(ClickHouseConfig c
     }
 
     private static NativeContext createNativeContext(ClickHouseConfig configure) throws SQLException {
-        NativeClient nativeClient = NativeClient.connect(configure);
+        if (configure.hosts().size() == 1) {
+            NativeClient nativeClient = NativeClient.connect(configure);
+            return new NativeContext(clientContext(nativeClient, configure), serverContext(nativeClient, configure), nativeClient);
+        }
+
+        return createFailoverNativeContext(configure);
+    }
+
+    private static NativeContext createFailoverNativeContext(ClickHouseConfig configure) throws SQLException {
+        NativeClient nativeClient = null;
+        SQLException lastException = null;
+
+        int tryIndex = 0;
+        do {
+            String hostAndPort = configure.hosts().get(tryIndex);
+            String[] hostAndPortSplit = hostAndPort.split(PORT_DELIMITER, 2);
+            String host = hostAndPortSplit[0];
+            int port;
+
+            if (hostAndPortSplit.length == 2) {
+                port = Integer.parseInt(hostAndPortSplit[1]);
+            } else {
+                port = configure.port();
+            }
+
+            try {
+                nativeClient = NativeClient.connect(host, port, configure);
+            } catch (SQLException e) {
+                lastException = e;
+            }
+            tryIndex++;
+        } while (nativeClient == null && tryIndex < configure.hosts().size());
```

Comment:
Looks like the former node always has high priority, so it does not a load-balanced implementation? If yes, let's add clear comments on the regex pattern to avoid surprising users.

Reply:
it does not a load-balanced implementation